rules/manager.go

```diff
@@ -403,3 +415,76 @@ func SendAlerts(s Sender, externalURL string) NotifyFunc {
 		}
 	}
 }
+
+// RuleConcurrencyController controls whether rules can be evaluated concurrently. Its purpose it to bound the amount
+// of concurrency in rule evaluations, to not overwhelm the Prometheus server with additional query load.
+// Concurrency is controlled globally, not on a per-group basis.
+type RuleConcurrencyController interface {
+	// RuleEligible determines if a rule can be run concurrently.
```

Comment:
```suggestion
	// RuleEligible determines if the rule can guarantee correct results while running concurrently.
```

Reply:
Addressed in separate PR where I'm picking up the work: https://github.com/prometheus/prometheus/pull/13476/commits/d0349dcac5e851b60a05c6f7298aa90592b50f67